client/question/services/FeedbackGeneratorServiceSpec.js

```diff
@@ -171,61 +167,131 @@ describe('FeedbackGeneratorService', function() {
   });
 
   describe('_getCorrectnessTestFeedback', function() {
-    it([
-      'should return feedback on a user function that ',
-      'doesn\'t pass all correctness tests'
-    ].join(''), function() {
-      var testCase = TestCaseObjectFactory.create({
-        input: 'cat',
-        allowedOutputs: ['a']
-      });
+    var sampleInputTestSuite = {
+      id: 'SAMPLE_INPUT',
+      testCase: {
+        input: 'Hello, John',
+        allowedOutputs: ['olleH, nhoJ']
+      }
+    };
+    var generalTestSuite = {
+      id: 'GENERAL_CASE',
+      testCase: {
+        input: 'Hi, world',
+        allowedOutputs: ['iH, dlrow']
+      }
+    };
+    var whitespaceTestSuite = {
+      id: 'WHITESPACE',
+      testCase: {
+        input: 'hello    ',
+        allowedOutputs: ['olleh    ']
+      }
+    };
+    var sampleInputTestCase;
+    var generalInputTestCase;
+    var whitespaceTestCase;
+
+    beforeEach(function() {
+      sampleInputTestCase =
+        TestCaseObjectFactory.create(sampleInputTestSuite.testCase);
+      generalInputTestCase =
+        TestCaseObjectFactory.create(generalTestSuite.testCase);
+      whitespaceTestCase =
+        TestCaseObjectFactory.create(whitespaceTestSuite.testCase);
+    });
 
-      var paragraphs = FeedbackGeneratorService
-        ._getCorrectnessTestFeedback(null, testCase, 'output').getParagraphs();
-      expect(paragraphs.length).toEqual(5);
-      expect(paragraphs[0].isTextParagraph()).toBe(true);
-      expect(paragraphs[0].getContent()).toBe(
-        "Your code produced the following result:");
-      expect(paragraphs[1].isCodeParagraph()).toBe(true);
-      expect(paragraphs[1].getContent()).toBe(
-        "Input: \"cat\"\nOutput: \"output\"");
-      expect(paragraphs[2].isTextParagraph()).toBe(true);
-      expect(paragraphs[2].getContent()).toBe(
-        "However, the expected output is:");
-      expect(paragraphs[3].isCodeParagraph()).toBe(true);
-      expect(paragraphs[3].getContent()).toBe("\"a\"");
-      expect(paragraphs[4].isTextParagraph()).toBe(true);
-      expect(paragraphs[4].getContent()).toBe(
-        "Could you fix this?");
+    it('should allow user to display output if suite id is \'SAMPLE_INPUT\'',
+      function() {
+        var correctnessFeedbackParagraphs =
+          FeedbackGeneratorService._getCorrectnessTestFeedback(
+          sampleInputTestCase, sampleInputTestSuite.id, 0,
+          'incorrect answer').getParagraphs();
+        expect(CORRECTNESS_FEEDBACK_TEXT.OUTPUT_ENABLED).toContain(
+          correctnessFeedbackParagraphs[0].getContent());
+        expect(correctnessFeedbackParagraphs[0].isTextParagraph()).toEqual(
+          true);
+        expect(correctnessFeedbackParagraphs[1].isOutputParagraph()).toEqual(
```

Comment:
Verify the content of this paragraph? Ditto in other tests below.

Reply:
Done